### ISSUE TITLE
caveman ninja doesnt work

### ISSUE THREAD

**Issue by stsp** [dosemu2/dosemu2#921]
**Describe the bug**
Crashes at startup.

**To Reproduce**
Just start the game.

**Attach the binaries or provide an URL**
https://www.myabandonware.com/game/joe-mac-caveman-ninja-16l

OK, dosemu2 can't run most games.

**Comment by jschwartzenberg** [comment]
This doesn't work in dosemu-1.5.0 either.

**Comment by stsp** [comment]
Yes, I didn't mark it as regression.

**Comment by jschwartzenberg** [comment]
I noticed, but just wanted to confirm :)

I suppose it's unclear what's broken.

**Comment by stsp** [comment]
I simply haven't looked yet. so unclear.

**Comment by stsp** [comment]
It works only with vm86sim sometimes.
Which likely means there is some timing problem.
Can you try it under qemu?

**Comment by jschwartzenberg** [comment]
Why qemu and not dosbox/pcem/something else? You specifically mean qemu with KVM?

**Comment by stsp** [comment]
Because qemu exhibits all those timing problems,
while dosbox/pcem emulate slow PC and things
will work there (i.e. why would I ask you to test
something for which the result is known beforehand).
As for with or w/o KVM - try whatever is easier to try.

**Comment by jschwartzenberg** [comment]
I just tried it again in DOSEMU2 and now it runs here. I didn't do anything special!

In qemu the game crashes. To quickly reproduce:
1. get the image from here: https://github.com/palmercluff/qemu-images/tree/master/freedos-1.2
2. unpack
3. run: `qemu-system-i386 -hda freedos-1.2.qcow2 -m 64 -hdb fat:${HOME}/.dosemu/drive_c/` (or point to wherever the game is, this example will cause dosemu's drive c to be available through `d:` in qemu.

**Comment by stsp** [comment]
Yes, it runs sometimes.

**Comment by jschwartzenberg** [comment]
Yes, you're right, it seems I was extremely lucky yesterday. Maybe this could be solved with some slow-down utility? Or does it even run when ran through DOS on hardware that's fast by itself already?

**Comment by stsp** [comment]
Now you understand why I asked
to test on qemu. :)

**Comment by stsp** [comment]
Is this still happening?

**Comment by jschwartzenberg** [comment]
Seems not anymore. I just get stuck at the copy protection screen now :)

**Comment by stsp** [comment]
Its no longer available for d/l
with the URL above. Now there
is "buy game on steam".
So please update URL.

**Comment by jschwartzenberg** [comment]
Here's the version I have: [joemac.tar.gz](https://github.com/dosemu2/dosemu2/files/10343194/joemac.tar.gz)

**Comment by stsp** [comment]
Works perfectly up to a
copy-protection screen.

**Comment by jschwartzenberg** [comment]
Yeah, let's close this. If it fails further on, that could be another bug.

**Comment by stsp** [comment]
Doesn't work for me again...
Very strange.
Same for you?

**Comment by jschwartzenberg** [comment]
It seems to depend on some random factors. The first time I tried it, it didn't work. The second time, it got to a graphical screen.

**Comment by stsp** [comment]
I think it does timer-based speed calibration.
Unfortunately the precise timers stack was
left half-written, so this will have to wait.
